### PR TITLE
Generic objects validation

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/38703

## Description
Adding validations to the new Generic objects entities (ObjectModules, ObjectTypes, ObjectFields): 
* Create 3 new schemes: objecttype and objectfield which are based on incidenttype and incidentfield schemes, and
   objectmodule scheme. 
* Integrating the validation of the new objects in the flow of demisto-sdk validate.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by roysagi — CHANGES_REQUESTED** [review]
@ShacharKidor - This looks good but please add tests.

@Itay4 - FYI - I think we should push this in order to push your example pack PR.

**Comment by Itay4** [comment]
> @ShacharKidor - This looks good but please add tests.
> 
> @Itay4 - FYI - I think we should push this in order to push your example pack PR.

ill test my pr with this branch

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -1105,6 +1107,14 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
                     return FileType.INCIDENT_FIELD
                 if _id.startswith('indicator'):
                     return FileType.INDICATOR_FIELD
+                else:  # not an incident or an indicator
+                    if 'definitionId' in _dict and 'genericModuleId' in _dict:
```
Comment (roysagi) [inline demisto_sdk/commands/common/tools.py:1111]: We might not save the `genericModuleId` field on content repo. so let's remove this condition (same for incident types above)

File: demisto_sdk/commands/common/tools.py [hunk 2]
```diff
@@ -1105,6 +1107,14 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
...
+                        if 'cliName' in _dict:
+                            return FileType.GENERIC_FIELD
```
Comment (roysagi) [inline demisto_sdk/commands/common/tools.py:1113]: let's move this before the recognition of Incident fields.
Let's check that the `definitionId` is not `incident` or `indicator` and if so this should be classified as Generic Field

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -1105,6 +1107,14 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
...
+                        elif 'color' in _dict:
+                            return FileType.GENERIC_TYPE
```
Comment (roysagi) [inline demisto_sdk/commands/common/tools.py:1115]: Same comment as Generic field detection, this time along side incident types.

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -1105,6 +1107,14 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
...
+                    elif 'definitions' in _dict and 'views' in _dict:
+                        return FileType.GENERIC_MODULE
```
Comment (roysagi) [inline demisto_sdk/commands/common/tools.py:1117]: Move this above line 1103 (before field identification) - only modules will have views and definitions

**Review by roysagi — CHANGES_REQUESTED** [review]
Let's add a changelog as well

**Comment by Itay4** [comment]
validation passes - https://app.circleci.com/pipelines/github/demisto/content/99603/workflows/fea3479d-a1cd-473b-8985-b9e7d3598eac/jobs/395819
note the following:
```
================= Running validation on changed pack unique files =================

Validating Packs/RBVM unique pack files

Running pack dependencies validation on RBVM

Couldn't find any items for pack 'RBVM'. Please make sure:
1 - The spelling is correct.
2 - The id_set.json file is up to date. Delete the file by running: `rm -rf Tests/id_set.json` and rerun the command.
No first level dependencies found
id set validations for: Packs/RBVM
```
@roysagi if im not wrong, this is expected

File: demisto_sdk/commands/validate/validate_manager.py [hunk 5]
```diff
@@ -226,6 +229,10 @@ def detect_file_level(file_path: str) -> PathLevel:
         if dir_name in CONTENT_ENTITIES_DIRS:
             return PathLevel.CONTENT_ENTITY_DIR
 
+        path_without_last_dir = file_path.rstrip(f'{dir_name}').rstrip('/')
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:232]: might be wiser to use:
`str(os.path.dirname(path)).endswith(GENERIC_TYPES_DIR)`

This way you can support windows devs as well.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 * Added a validation that mapper's id and name are matching. Updated the format of mapper to include update_id too.
 * Added a validation to ensure that image paths in the README files are valid.
 * Fixed **find_type** function to correctly find test files, such as, test script and test playbook.
+* Added scheme validations to the new Generic Objects.
```
Comment (roysagi) [inline CHANGELOG.md:8]:
```suggestion
* Added scheme validations for the new Generic Object Types, Fields, and Modules.
```

**Review by roysagi — APPROVED** [review]

**Pull request merged**